src/styles/animal-list.scss

```diff
@@ -258,8 +258,10 @@ $dark-orange: #f60;
   font-weight: 500;
   font-size: 16px;
   color: $dark-grey;
-  flex: 1;
+  flex: 3;
```

Comment:
Usa flex definiendo las 3 propiedades, asi no dejas nada librado al azar, esto es generalmente lo que rompe safari.
https://css-tricks.com/snippets/css/a-guide-to-flexbox/

Seguramente: `flex: 3 1 auto;` te funcione, pero probalo